### PR TITLE
cache nodes from search results

### PR DESCRIPTION
**Pull request by icewind1991**
Saves a query for every file in the search result

### CODE REVIEW

File: apps/dav/lib/Connector/Sabre/Server.php [hunk 1]
```diff
@@ -33,6 +33,7 @@
  * @see \Sabre\DAV\Server
  */
 class Server extends \Sabre\DAV\Server {
+	/** @var CachingTree $tree */
```
Comment (blizzz) [inline apps/dav/lib/Connector/Sabre/Server.php]: out of place?

File: apps/dav/lib/Connector/Sabre/CachingTree.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	 * @param Node $node
+	 * @param null|string $path
+	 */
+	public function cacheNode(Node $node, $path = null) {
+		if (is_null($path)) {
+			$path = $node->getPath();
+		}
+		$this->cache[trim($path, '/')] = $node;
+	}
+}
```
Comment (blizzz) [inline apps/dav/lib/Connector/Sabre/CachingTree.php]: add EOL

**Review by blizzz — COMMENTED** [review]
Code looks OK!

File: apps/dav/lib/Connector/Sabre/CachingTree.php (hunk 2 above)
Comment (icewind1991) [inline apps/dav/lib/Connector/Sabre/CachingTree.php]: done

File: apps/dav/lib/Connector/Sabre/Server.php (hunk 1 above)
Comment (icewind1991) [inline apps/dav/lib/Connector/Sabre/Server.php]: This overwrites the type hit for `$tree` which is defined in the parent class

**Comment by blizzz** [comment]
some tests are failing

**Comment by icewind1991** [comment]
Fixed the tests

**Comment by icewind1991** [comment]
@blizzz @MorrisJobke @rullzer please review

**Review by blizzz — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
Code looks good, tested and is a little bit faster for big search result lists.

**Pull request merged**